test/helpers.ts

```diff
@@ -26,3 +26,17 @@ export function verify(result: IGitResult, callback: (result: IGitResult) => voi
     throw e
   }
 }
+
+/**
+ * Returns with the container directory for the Git executable.
+ * This function could come handy, when trying to set the `LOCAL_GIT_DIRECTORY` environment variable
+ * for `dugite`. Passing into the exact path of the Git executable will return with the local Git directory
+ * path that is expected and consumed by `dugite`.
+ *
+ * @param gitPath the FS path to the Git executable.
+ */
+export function getGitDirPath(gitPath: string): string {
```

Comment:
I'm curious why we can't use [`path.dirname()`](https://nodejs.org/api/path.html#path_path_dirname_path) here to get the directory associated with the Git executable.

Reply:
I am glad you've asked, I do not like that function either so that we can discuss.

When we [resolve the Git binary](https://github.com/desktop/dugite/blob/master/lib/git-environment.ts#L26) in the `git-environment` module, we expect not the direct container folder of the executable but one level up. The `find-git-exec` library, I have written, provides the exact executable path; hence we need to traverse up two levels instead of one. Unfortunately, due to the lack of my JS/TS knowledge, I did not have a better idea how to go up two levels. Does this clarify my changes?

I look forward to your feedback, and if you have a better approach, I am happy to adjust my PR. Thank you!

Reply:
I got it what you meant. I will adjust it right now.